salt/modules/file.py

```diff
@@ -2093,14 +2093,10 @@ def replace(path,
                         # Content was found, so set found.
                         found = True
 
-                # Keep track of show_changes here, in case the file isn't
-                # modified
-                if show_changes or append_if_not_found or \
-                   prepend_if_not_found:
-                    orig_file = r_data.read(filesize).splitlines(True) \
-                        if isinstance(r_data, mmap.mmap) \
-                        else r_data.splitlines(True)
-                    new_file = result.splitlines(True)
+                orig_file = r_data.read(filesize).splitlines(True) \
```

Comment:
Sorry - are you suggesting it was broken on purpose to save memory? Do we have such a warning some place when using `show_changes` (and if not, what's the difference)?